website/routes.py

```diff
@@ -183,6 +190,27 @@ def __init__(self, *args, **kwargs):
         kwargs['data'] = get_globals
         super(OsfWebRenderer, self).__init__(*args, **kwargs)
 
+    def __call__(self, data, *args, **kwargs):
+        """
+        This function has been added to keep our Flask requests compatible with django-waffle, it's been adapted from
+        waffle's own middleware code at https://github.com/django-waffle/django-waffle/blob/master/waffle/middleware.py
+        """
+
+        resp = super(OsfWebRenderer, self).__call__(data, *args, **kwargs)
+        secure = get_setting('SECURE')
+        max_age = get_setting('MAX_AGE')
+
+        if hasattr(request, 'waffles'):
+            for k in request.waffles:
+                name = smart_str(get_setting('COOKIE') % k)
+                active, rollout = request.waffles[k]
+                if rollout and not active:
+                    # "Inactive" is a session cookie during rollout mode.
+                    age = None
+                else:
+                    age = max_age
+                resp.headers.add('Set-Cookie', dump_cookie(name.encode(), bytes(active), age, bytes(secure)))
```

Comment:
Why doesn't `resp.set_cookie(name.encode(), value=bytes(active), max_age=age, secure=bytes(secure))` work? I didn't dig into it further -

Reply:
I think I tried this and it overrid other cookies, or it caused some other problem.

Reply:
ok, if it is overriding other cookies, obviously don't want to do this, but I wonder if you weren't just passing in the arguments in the right format